### DISCUSSION TITLE
extract_words() behaving poorly while extract_text() works fine with the same parameters.

### DISCUSSION THREAD

**Discussion by PrimoJefe** [jsvine/pdfplumber#503] [Q&A] [answered]
When I use extract_text() it prints out fine as you can see here :

![extract_text](https://user-images.githubusercontent.com/56933843/132760374-788cb87d-6158-47bf-8857-d0b09bbdc5c2.png)

With every other pdfs for now I did not have any problems, but in this pdf when using extract_words() on the same part it includes white spaces : 

![extract_words](https://user-images.githubusercontent.com/56933843/132760514-7c325299-4b9a-43d5-b584-d1e827d2e007.png)

At first I thought I needed to change the tolerance but while checking the values of x0 and x1 those white spaces are not supposed to be detected, unless there is something I did not understand. Here are the values :

![Example](https://user-images.githubusercontent.com/56933843/132759429-cf667638-8f66-42ea-bbd0-7dc3c2ed397c.png)

Anyone has any idea what is going on? 
Thanks in advance

**Comment by jsvine** [comment, marked as answer]
Hello, and thanks for your interest in this library. Apologies for not responding sooner. Are you able to provide the PDF and specific page number? It's a bit difficult to debug without it. But a couple of observations that may or may not be helpful:

- You're passing a `y_tolerance` in your first example, but not in your second. 

- Passing `extra_attrs` changes how `pdfplumber` groups characters into words. In your example, if two characters do not share the same `size` and `fontname`, they will not be grouped into the same word.

**Reply by PrimoJefe** [reply]
Hi, thank you for your answer.

Indeed, the problem was that some characters inside the same words did not have the same fontname for some reason.
I did not expect that behavior, also they were pretty similar (BHKDKN+stoneSans vs BHJKJD+stoneSans). 

Thank you again for your time, this library is great.
